core/payment-driver/base/src/utils.rs

```diff
@@ -0,0 +1,35 @@
+/*
+    Common utility functions for dealing with PaymentDriver related objects
+*/
+
+// External crates
+use chrono::Utc;
+use serde_json;
+
+// Workspace uses
+use ya_core_model::driver::{GenericError, PaymentConfirmation, PaymentDetails, SchedulePayment};
+
+pub fn to_payment_details(msg: SchedulePayment) -> PaymentDetails {
+    PaymentDetails {
+        recipient: msg.recipient().to_string(),
+        sender: msg.sender().to_string(),
+        amount: msg.amount(),
+        date: Some(Utc::now()),
+    }
+}
```

Comment:
~I think it would be better to implement `From<SchedulePayment>` for `PaymentDetails` in `ya_core_model`.~
On second thoughts... this looks easy to misuse. The problem is with `date`. It should be the moment when the payment has been made and this has no guarantee to be called in that moment. I believe it should accept `date` as parameter.

Reply:
I like the from idea, can it also take a date?

If not i'll change this to take date as argument

Reply:
You can `impl From<(SchedulePayment, DateTime<Utc>)>` but I think it's ugly. Better just add a param.